### ISSUE TITLE
`function-paren-newline`: add a case for function taking a callback as the single argument

### ISSUE THREAD

**Issue by fregante** [eslint-stylistic/eslint-stylistic#1191]
> [!NOTE]
> This is a copy-paste of https://github.com/eslint-stylistic/eslint-stylistic/issues/290, which appears to have been mention-closed incorrectly 

### Clear and concise description of the problem

Take the following code : 
```ts
const array = [ 'bob', 'billy' ];
array.find(arrayItem => 
  arrayItem.startsWith('bob')
);
```

There is no configuration of `@stylistic/function-paren-newline` that would keep this code formatted that way, with a line break after the arrow (with or without curlies) and one before the last parenthesis :
- 'consistent' or 'always' expects a break before the start of the predicate to keep the last paren on its line
- 'multiline' and 'multiline-arguments' doesn't apply to arrow function bodies

### Suggested solution

A new option ('callback-only' ?) could be added that allows/expect a line break before the closing paren of a function taking a single callback argument.

### Alternative

I tried all the other options of `function-paren-newline` but the only way to allow this style is disabling the rule (which I don't want to do)

### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### Contributes

- [ ] If this feature request is accepted, I am willing to submit a PR to fix this issue
